### ISSUE TITLE
Blank Login Screen (Arch Linux)

### ISSUE THREAD

**Issue by MarcPG1905** [KRTirtho/spotube#1817]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I just installed spotube over the arch user repositories (pretty sure the `spotube-bin` one) and whenever trying to log in using spotify, all I get is an empty white window with three buttons, which also just do nothing.
![Screenshot of empty Window](https://github.com/user-attachments/assets/4059c1b6-a9c4-4d4e-a2ce-0fc20d1a283b)

### Expected Behavior

It should just open the login page and I should be able to log in and use the program.

### Steps to reproduce

1. Install Spotube (`spotube-bin`) over the arch user repositories using `yay -Sy spotube-bin`.
2. Open Spotube (obviously).
3. Try to login in the settings menu.
4. Restart and try a few more times, without success.

### Operating System

Arch Linux (6.10.6-arch1-1)

### Spotube version

v3.8.0

### Installation source

Arch User Repository (AUR)

### Additional information

My arch installation is running on GNOME and I have an RTX 4070 running using the latest nvidia-dmks drivers.

### Self grab

- [ ] I'm ready to work on this issue!

**Comment by dipzza** [comment]
Same issue and OS, but using flatpak as the installation source

**Comment by MarcPG1905** [comment]
I just checked again and it seems like it's just not rendering. I can still click some things in the white box and my cursor also changes. I just have no way of knowing what I'm clicking or what to do.
Probably something graphics related?

**Comment by miguelpecanha** [comment]
I am using Artix Linux Cinnamon and having the same problem

**Comment by Goaty1208** [comment]
Using arch and I have the same issue with the source, flatpak and aur versions.

**Comment by Underio3298** [comment]
Same issue with all versions, using Arch linux with XFCE.

**Comment by wuniversales** [comment]
I have not had that problem in Manjaro with the flatpak version, it is true that just opened the program did not load the login, but when I closed the program and reopened it, from the program settings allowed me to connect the account without problems.

**Comment by PieroganoReggiano** [comment]
There is a workaround here: https://github.com/DioxusLabs/dioxus/issues/1909

TLDR open spotube with env variable WEBKIT_DISABLE_COMPOSITING_MODE=1

**Comment by MarcPG1905** [comment]
> There is a workaround here: [DioxusLabs/dioxus#1909](https://github.com/DioxusLabs/dioxus/issues/1909)
> 
> TLDR open spotube with env variable WEBKIT_DISABLE_COMPOSITING_MODE=1

That seems to have fixed the issue, but I still think this issue should stay open, because that isn't really a user-friendly workaround and there shouldn't be the need for a workaround at all.

**Comment by colonelpopcorn** [comment]
> There is a workaround here: [DioxusLabs/dioxus#1909](https://github.com/DioxusLabs/dioxus/issues/1909)
> 
> TLDR open spotube with env variable WEBKIT_DISABLE_COMPOSITING_MODE=1

This worked for me, but should probably be pinned until the underlying bug can be fixed. For users who don't want to launch spotube from the CLI you can edit the `spotube.desktop` file (typically) found in `/usr/share/applications/` directory. On the line that shows `Exec=spotube %U` you can add the variable setting like so `Exec=WEBKIT_DISABLE_COMPOSITING_MODE=1 spotube %U`.

Re: The bug, I did try updating the desktop_webview_window plugin with the latest from upstream@MixinNetwork but it did not solve the issue. I then did a lot of searching and found that other people were having similar issues with webkitgtk2. There's a handful of issues with other apps (collected [here](https://github.com/tauri-apps/tauri/issues/9304)) that point to an incompatibility with Nvidia GPUs (I have a 1060) and webkitgtk2. I didn't want to accept that because all the webkit/gtk bug reports about errors on Nvidia GPUs have been fixed since at least February of this year (2024). So I checked to see what versions of libwebkit2gtk I had installed and while it's not a lot it's more than one version.

```txt
libwebkit2gtk-4.0-doc/noble-updates,noble-updates 2.44.2-0ubuntu0.24.04.2 all
libwebkit2gtk-4.1-0/noble-updates,now 2.44.2-0ubuntu0.24.04.2 amd64 [installed]
libwebkit2gtk-4.1-0/noble-updates 2.44.2-0ubuntu0.24.04.2 i386
libwebkit2gtk-4.1-dev/noble-updates,now 2.44.2-0ubuntu0.24.04.2 amd64 [installed]
libwebkit2gtk-4.1-dev/noble-updates 2.44.2-0ubuntu0.24.04.2 i386
webkit2gtk-driver/noble-updates 2.44.2-0ubuntu0.24.04.2 amd64
webkit2gtk-driver/noble-updates 2.44.2-0ubuntu0.24.04.2 i386
```

It's likely an error in webkit2gtk or one of its dependencies. :man_shrugging:

**Comment by Burtfaceman** [comment]
This bug still seems to be present. I just installed Spotube via Flatpak on Linux Mint DE 6, and I get a blank login screen like @MarcPG1905's screenshot.

**Comment by HYDRAWOLFY** [comment]
I am on Ubuntu 22.04 LTS and encountered the same problem.I am new to linux eco system and this really hinders my productivity (not being able to listen to music)I hope this gets fixed as soon as possible.I also have an Nvidia GPU(GT 710,running with the open source drivers)

**Comment by MarcPG1905** [comment]
> I am on Ubuntu 22.04 LTS and encountered the same problem.I am new to linux eco system and this really hinders my productivity (not being able to listen to music)I hope this gets fixed as soon as possible.I also have an Nvidia GPU(GT 710,running with the open source drivers)

You can always just use the fix mentioned above until they fix it. Just do `WEBKIT_DISABLE_COMPOSITING_MODE=1 spotube` over the console and the log in. You only need to start it over the console with the custom command once, to log in, and then you won't need to do it afterwards because you don't need to log in again.

**Comment by HYDRAWOLFY** [comment]
Um which console? (Kinda confused)Since I am new

**Comment by MarcPG1905** [comment]
> Um which console? (Kinda confused)Since I am new

Like the normal console/terminal/shell or whatever you call it. Depending on your desktop it will be named differently.
Honestly not sure if this question is a joke or not because even windows has cmd

**Comment by HYDRAWOLFY** [comment]
Nah i thought I had to go into cookies and stuff and inject code there in the window itself and not in the terminal.lol sorry if I sounded stupid

**Comment by fareed-hameed** [comment]
Hi. I have the same issue as well in Android 9. I tried it on another device, and it works. So it is random, but it is still persistent. I am not sure if this workaround can be used in an unrooted Android device?

**Comment by Gater73** [comment]
Still happening in Arch linux with KDE on wayland (AMD GPU)

**Comment by MarcPG1905** [comment]
Version 4.0.0-1 has fixed this issue.

For anyone using older versions, `WEBKIT_DISABLE_COMPOSITING_MODE=1` is still a working workaround, but I strongly advise just updating to 4.0.0.

**Comment by ykcgw** [comment]
On ArchLinux I did a script and it works:

```
#!/bin/bash

export WEBKIT_DISABLE_COMPOSITING_MODE=1
/usr/bin/spotube
```
I'm on Nvidia drivers.
